src/main/java/jenkins/branch/DeadBranchStrategyDescriptor.java

```diff
@@ -53,12 +53,9 @@ public boolean isApplicable(java.lang.Class<? extends MultiBranchProject> jobTyp
     @SuppressWarnings("unused") // used by stapler
     public static List<DeadBranchStrategyDescriptor> forProject(Class<? extends MultiBranchProject> jobType) {
         List<DeadBranchStrategyDescriptor> result = new ArrayList<DeadBranchStrategyDescriptor>();
-        for (Descriptor<DeadBranchStrategy> d : Jenkins.getInstance().getDescriptorList(DeadBranchStrategy.class)) {
-            if (d instanceof DeadBranchStrategyDescriptor) {
-                DeadBranchStrategyDescriptor descriptor = (DeadBranchStrategyDescriptor) d;
-                if (descriptor.isApplicable(jobType)) {
-                    result.add(descriptor);
-                }
+        for (DeadBranchStrategyDescriptor descriptor : ExtensionList.lookup(DeadBranchStrategyDescriptor.class)) {
```

Comment:
Nope, those two fields are merely cached views over the underlying set of `ExtensionComponent`s, which the plugin system treats as singletons.

From `/script` on `hpi:run` of this project:

```groovy
println(ExtensionList.lookup(jenkins.branch.DeadBranchStrategyDescriptor));
println(Jenkins.instance.getDescriptorList(jenkins.branch.DeadBranchStrategy));
```

→

```
[jenkins.branch.DefaultDeadBranchStrategy$DescriptorImpl@1e1aaec5]
[jenkins.branch.DefaultDeadBranchStrategy$DescriptorImpl@1e1aaec5]
```